lib/features/paywall/domain/model/paywall_url_pattern.dart

```diff
@@ -0,0 +1,40 @@
+import 'package:core/utils/mail/domain.dart';
+import 'package:core/utils/mail/mail_address.dart';
+import 'package:equatable/equatable.dart';
+import 'package:get/get.dart';
+import 'package:tmail_ui_user/features/paywall/presentation/paywall_utils.dart';
+
+class PaywallUrlPattern with EquatableMixin {
+  final String pattern;
+
+  PaywallUrlPattern(this.pattern);
+
+  String getQualifiedUrl({required String ownerEmail, String? domainName}) {
```

Comment:
`domainName` still not in use. IMO, we should use it, because not all the time domain of email is the same with domain of platform